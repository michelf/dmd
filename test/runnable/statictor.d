// PERMUTE_ARGS:
// POST_SCRIPT: runnable/extra-files/statictor-postscript.sh

private import std.stdio;

class Foo
{
        static this() {printf("Foo static ctor\n");}
        static ~this() {printf("Foo static dtor\n");}
}

static this() {printf("static ctor\n");}
static ~this() {printf("static dtor\n");}

shared static this()
{
    printf("shared static this()\n");
}

shared static ~this()
{
    printf("shared static this()\n");
}

class Bar
{
        static this() {printf("Bar static ctor\n");}
        static ~this() {printf("Bar static dtor\n");}
}

void main()
{
}

