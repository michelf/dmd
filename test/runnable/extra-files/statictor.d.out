shared static this()
Foo static ctor
static ctor
Bar static ctor
Bar static dtor
static dtor
Foo static dtor
shared static this()
